File: Codigos/Ejercicio4.java
import java.util.ArrayList;
import java.util.Scanner;
 
public class Ejercicio4 {
    static ArrayList<String> datos = new ArrayList<>();
    static Scanner sc = new Scanner(System.in);
 
    static void mostrar() {
        if (datos.isEmpty()) { System.out.println("(Lista vacía)"); return; }
        for (int i = 0; i < datos.size(); i++)
            System.out.println(i + ". " + datos.get(i));
    }
 
    public static void main(String[] args) {
        int opcion;
        do {
            System.out.println("\n=== MENU CRUD ===");
            System.out.println("1.Agregar 2.Mostrar 3.Actualizar 4.Eliminar 0.Salir");
            System.out.print("Opcion: ");
            opcion = sc.nextInt();
 
            if (opcion == 1) {
                System.out.print("Ingrese item: ");
                datos.add(sc.next());
                System.out.println("Agregado correctamente.");
            } else if (opcion == 2) {
                mostrar();
            } else if (opcion == 3) {
                mostrar();
                System.out.print("Posicion a actualizar: ");
                int pos = sc.nextInt();
                if (pos >= 0 && pos < datos.size()) {
                    System.out.print("Nuevo valor: ");
                    datos.set(pos, sc.next());
                } else System.out.println("Posicion invalida.");
            } else if (opcion == 4) {
                mostrar();
                System.out.print("Posicion a eliminar: ");
                int pos = sc.nextInt();
                if (pos >= 0 && pos < datos.size()) datos.remove(pos);
                else System.out.println("Posicion invalida.");
            }
        } while (opcion != 0);
        sc.close();
    }
}

File: Codigos/Ejercicio4.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
 
void mostrar(const vector<string>& lista) {
    if (lista.empty()) { cout << "(Lista vacía)" << endl; return; }
    for (int i = 0; i < lista.size(); i++)
        cout << i << ". " << lista[i] << endl;
}
 
int main() {
    vector<string> datos;
    int opcion; string item;
 
    do {
        cout << "\n=== MENU CRUD ===" << endl;
        cout << "1. Agregar "<< endl;
        cout << "2. Mostrar "<< endl; 
        cout << "3. Actualizar" << endl;
        cout << "4. Eliminar" << endl;
        cout << "0. Salir" << endl;
        cout << "Opcion: "; cin >> opcion;
 
        if (opcion == 1) {
            cout << "Ingrese item: "; cin >> item;
            datos.push_back(item);
            cout << "Agregado correctamente." << endl;
        } else if (opcion == 2) {
            mostrar(datos);
        } else if (opcion == 3) {
            mostrar(datos);
            int pos; cout << "Posicion a actualizar: "; cin >> pos;
            if (pos >= 0 && pos < datos.size()) {
                cout << "Nuevo valor: "; cin >> item;
                datos[pos] = item;
            } else cout << "Posicion invalida." << endl;
        } else if (opcion == 4) {
            mostrar(datos);
            int pos; cout << "Posicion a eliminar: "; cin >> pos;
            if (pos >= 0 && pos < datos.size())
                datos.erase(datos.begin() + pos);
            else cout << "Posicion invalida." << endl;
        }
    } while (opcion != 0);
    return 0;
}
